azure/azure.go

```diff
@@ -89,20 +94,46 @@ type Config struct {
 	UserAgent          string
 }
 
-// calculateMidThroughputUltra calculates the median throuput for a particular IOPS value for Ultra Disks
-func calculateMidThroughputUltra(iops int64) (maxThroughput int64) {
-	return int64(math.Min(float64(iops*128/1024), maxThroughputUltra)) // Convert kB/s to MB/s
+// updateUltraIopsThroughput - validates if the requested IOPS and throuput are in range - If not update with minimum
+func updateUltraIopsThroughput(size int32, reqIops, reqTP *int64) {
+	minAllowedIOPS := int64(math.Max(minIopsUltra, float64(size)))
+	maxAllowedIOPS := int64(math.Min(maxIopsUltra, float64(size*300)))
+	if reqIops != nil {
+		if *reqIops < minAllowedIOPS || *reqIops > maxAllowedIOPS {
+			logrus.Warnf("UltraDisk : Requested IOPS: [%v] not in range for size: [%v] - defaulting to minimum iops: [%v]", *reqIops, size, minAllowedIOPS)
+			*reqIops = minAllowedIOPS
+		}
+		minAllowedTP := int64(math.Max(minThroughputUltra, math.Ceil(float64(*reqIops)*4/1024)))
+		maxAllowedTP := int64(math.Min(maxThroughputUltra, float64(*reqIops*256/1024)))
+		if reqTP != nil && (*reqTP < minAllowedTP || *reqTP > maxAllowedTP) {
+			logrus.Warnf("UltraDisk : Requested throughput [%v] not in range for iops [%v] - defaulting to minimum throughput : [%v]", *reqTP, *reqIops, minAllowedTP)
+			*reqTP = minAllowedTP
+		}
```

Comment:
As disucssed previously, do we want to do:
if reqTP < minAllowed -> reqTP = minAllowed.
if reqTP > maxAllowed -> reqTP = maxAllowed?